service/worker/failovermanager/workflow_test.go

```diff
@@ -39,15 +41,35 @@ import (
 	"go.uber.org/cadence/testsuite"
 )
 
+var clusters = []*types.ClusterReplicationConfiguration{
+	{
+		ClusterName: "c1",
+	},
+	{
+		ClusterName: "c2",
+	},
+}
+
 type failoverWorkflowTestSuite struct {
 	suite.Suite
 	testsuite.WorkflowTestSuite
+	activityEnv *testsuite.TestActivityEnvironment
+	workflowEnv *testsuite.TestWorkflowEnvironment
 }
 
 func TestFailoverWorkflowTestSuite(t *testing.T) {
 	suite.Run(t, new(failoverWorkflowTestSuite))
 }
 
+func (s *failoverWorkflowTestSuite) SetupTest() {
+	s.activityEnv = s.NewTestActivityEnvironment()
+	s.workflowEnv = s.NewTestWorkflowEnvironment()
+	s.workflowEnv.RegisterWorkflowWithOptions(FailoverWorkflow, workflow.RegisterOptions{Name: WorkflowTypeName})
+	s.workflowEnv.RegisterActivityWithOptions(FailoverActivity, activity.RegisterOptions{Name: failoverActivityName})
+	s.workflowEnv.RegisterActivityWithOptions(GetDomainsActivity, activity.RegisterOptions{Name: getDomainsActivityName})
+	s.activityEnv.RegisterActivityWithOptions(FailoverActivity, activity.RegisterOptions{Name: failoverActivityName})
+	s.activityEnv.RegisterActivityWithOptions(GetDomainsActivity, activity.RegisterOptions{Name: getDomainsActivityName})
+}
```

Comment:
Add a `TearDownTest()` and assert all expectations for mock inside it?

Reply:
All mocks are in cadence testing framework

Reply:
Yeah, I understand that. but we still need to call `env.AssertExpectations()` to check if all mocks are indeed used. like this: https://github.com/uber-common/cadence-samples/blob/master/cmd/samples/recipes/branch/workflow_test.go#L29

Reply:
Sounds good. Added it.